numpy/core/tests/test__exceptions.py

```diff
@@ -56,3 +57,38 @@ class TestUFuncNoLoopError:
     def test_pickling(self):
         """ Test that _UFuncNoLoopError can be pickled """
         assert isinstance(pickle.dumps(_UFuncNoLoopError), bytes)
+
+
+@pytest.mark.parametrize("args", [
+    (2, 1, None),
+    (2, 1, "test_prefix"),
+    (2, None, None),
+    (2, None, "test_prefix")
```

Comment:
These aren't valid call signatures.

Reply:
Ah, that makes more sense actually. I was wondering what exactly was going on with this `if`/`else` clause.

Anyway, fixed as of https://github.com/numpy/numpy/pull/19459/commits/2266599de3d13434b4e53c5c8580cb57bd969ee9. 
Both `ndim` and `axis` attributes will now default to `None` if a custom exception message is passed.